R/pkg/R/mllib.R

```diff
@@ -896,9 +898,10 @@ setMethod("summary", signature(object = "LogisticRegressionModel"),
 #' summary(savedModel)
 #' }
 #' @note spark.mlp since 2.1.0
-setMethod("spark.mlp", signature(data = "SparkDataFrame"),
-          function(data, layers, blockSize = 128, solver = "l-bfgs", maxIter = 100,
+setMethod("spark.mlp", signature(data = "SparkDataFrame", formula = "formula"),
+          function(data, formula, layers, blockSize = 128, solver = "l-bfgs", maxIter = 100,
```

Comment:
No, we must have ```formula```. The original PR is very tricky and even can be regarded as bug. We suppose the feature and label column as "features/label", but it should allow users to specify their own feature and label column by R formula. You can refer that almost all ML wrappers support R formula.